docs/aws/audit/redshiftmonitoring/rules/snapshot_retention_period.mdx

```diff
@@ -23,6 +23,209 @@ SOC2, HIPAA, GDPR, NISTCSF, PCIDSS
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can run any AWS CLI commands, you need to install and configure AWS CLI on your local system. You can do this by following the instructions provided in the AWS CLI User Guide.
+
+2. List all Redshift clusters: Use the following AWS CLI command to list all the Redshift clusters in your AWS account.
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+   ```
+   This command will return a list of all Redshift cluster identifiers.
+
+3. Describe each Redshift cluster: For each Redshift cluster identifier returned by the previous command, use the following AWS CLI command to describe the cluster and get its automated snapshot retention period.
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <ClusterIdentifier> --query 'Clusters[*].[AutomatedSnapshotRetentionPeriod]' --output text
+   ```
+   Replace `<ClusterIdentifier>` with the actual cluster identifier. This command will return the automated snapshot retention period for the specified Redshift cluster.
+
+4. Check the automated snapshot retention period: If the automated snapshot retention period returned by the previous command is 0, it means that the automated snapshots are not retained and the Redshift cluster is misconfigured. If the automated snapshot retention period is greater than 0, it means that the automated snapshots are retained for the specified number of days and the Redshift cluster is correctly configured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and establish a session with AWS using your access and secret keys.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a Redshift client using the session created above.
+
+```python
+redshift_client = session.client('redshift')
+```
+
+4. Use the `describe_clusters` method to retrieve information about all Redshift clusters. Then, iterate over each cluster and check the `AutomatedSnapshotRetentionPeriod` attribute. If the value is 0, it means the retention period is not enabled.
+
+```python
+clusters = redshift_client.describe_clusters()
+
+for cluster in clusters['Clusters']:
+    if cluster['AutomatedSnapshotRetentionPeriod'] == 0:
+        print(f"Cluster {cluster['ClusterIdentifier']} does not have retention period enabled for automated snapshots.")
+```
+
+This script will print out the identifiers of all Redshift clusters that do not have a retention period enabled for automated snapshots.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>